testing-unit-cs-top-level-program/README.md

```diff
@@ -0,0 +1,78 @@
+# C# Unit Testing with Top-Level Statements
+
+In this example project, we examine how to unit test C# Pulumi programs that are using top-level statements with dotnet SDK v6. These programs look like this
+
+```csharp
+using Pulumi;
+
+return await Deployment.RunAsync(() => 
+{
+    // Create resources here
+});
+```
+
+In order to unit test this piece of code, we have to do a bit of rewriting such that we don't run the `Deployment.RunAsync` function because it expects the program is being executed from the Pulumi CLI. 
+
+Now, we take the lambda that is provided to `Deployment.RunAsync` and move it in a place where we can reference it later from our unit test project. The following snippet shows this
+
+```csharp
+using Pulumi;
+
+return await Deployment.RunAsync(Deploy.Infra);
+
+class Deploy
+{
+    public static Dictionary<string, object?> Infra()
+    {
+        // Create resources here
+    }
+}
+```
+Here, the code that creates resources and returns outputs is inside the `Deploy.Infra` function which we can reference from the test project. 
+
+> The return type of `Deploy.Infra` could also have been any of the following 
+> - `void` when we are not returning outputs
+> - `Task` when we want to `await` async code
+> - `Task<Dictionary<string, object?>>` when we want to `await` async code _and_ return outputs
+
+There is a catch before we can reference `Deploy.Infra` from the unit tests. Since we don't have an explicit namespace here, we need to add this property to the project file
```

Comment:
eh this isn't due to namespace, this is because the default visibility for a class is "internal". It might be simpler to make the Deploy class public rather than getting into project changes.

Reply:
I thought the `public` modifier was not allowed in all of the top-level C# file but it was only for top-level functions so indeed I could mark the class `public` and use from the unit test project without making internals visible. Thanks for the nice catch!